src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelEventArgs.cs

```diff
@@ -153,4 +159,142 @@ public OpenDocumentationLinkEventArgs(Uri link)
             IsRemoteResource = link.IsAbsoluteUri && !link.IsFile;
         }
     }
+
+    /// <summary>
+    /// A node documentation request compatible with the Dynamo RequestOpenDocumentationLink event,
+    /// such as the namespace of the node used to lookup the .md documentation file
+    /// and additional Node Info gathered from the NodeModel properties.
+    /// </summary>
+    public class OpenNodeAnnotationEventArgs : OpenDocumentationLinkEventArgs
+    {
+        public string Namespace { get; }
+        public string Type { get; }
+        public string Description { get; }
+        public string Category { get; }
+        public List<string> InputNames { get; private set; }
+        public List<string> OutputNames { get; private set; }
+        public List<string> InputDescriptions { get; private set; }
+        public List<string> OutputDescriptions { get; private set; }
+
+        public OpenNodeAnnotationEventArgs(NodeModel model, DynamoViewModel dynamoViewModel) : base(new Uri(String.Empty,UriKind.Relative))
+        {
+            if (model == null) throw new ArgumentNullException(nameof(model));
+
+            Namespace = GetNamespace(model, dynamoViewModel);
+            Type = model.Name;
+            Description = model.Description;
+            Category = model.Category;
+            SetInputs(model);
+            SetOutputs(model);
+
+        }
+
+        private void SetOutputs(NodeModel nodeModel)
+        {
+            OutputNames = new List<string>();
+            OutputDescriptions = new List<string>();
+
+            var outputs = nodeModel.OutPorts;
+            foreach (var output in outputs)
+            {
+                OutputNames.Add(output.Name);
+                OutputDescriptions.Add(output.ToolTip);
+            }
+        }
+
+        private void SetInputs(NodeModel nodeModel)
+        {
+            InputNames = new List<string>();
+            InputDescriptions = new List<string>();
+
+            var inputs = nodeModel.InPorts;
+            foreach (var input in inputs)
+            {
+                InputNames.Add(input.Name);
+                InputDescriptions.Add(input.ToolTip);
+            }
+        }
+
+        static private string GetNamespace(NodeModel nodeModel, DynamoViewModel viewModel)
```

Comment:
this function is called `GetNameSpace` but it seems like it returns the `namespace.FunctionName` and sometimes other names...

Maybe... GetMinimumQualifiedName? or something like that?